src/main/java/org/reaktivity/specification/http/internal/Functions.java

```diff
@@ -38,13 +42,21 @@ public static String loginBase64Encoder(String login)
     public static byte[] header(String name, String value)
     {
         int nameLength = name.length(), valueLength = value.length();
+        byte[] nameBytes = name.getBytes(), valueBytes = value.getBytes();
         int length = nameLength + valueLength + 3;
         byte[] result = new byte[length];
-        result[0] = (byte) 0x00;
-        result[1] = (byte) nameLength;
-        System.arraycopy(name.getBytes(), 0, result, 2, nameLength);
-        result[2 + nameLength] = (byte) valueLength;
-        System.arraycopy(value.getBytes(), 0, result, 3 + nameLength, valueLength);
+
+        ByteBuffer buf = ByteBuffer.allocate(length);
+        buf.put((byte) 0x00);
+        buf.put((byte) nameLength);
+        buf.put(nameBytes);
+        buf.put((byte) valueLength);
+        buf.put(valueBytes);
```

Comment:
No need to explicitly layout fields, this is done for you by the flyweight builder.